pkg/commands/release.go

```
@@ -1,46 +1,408 @@
+
+		// Create artifacts array for this operator set
+		var artifacts []releasemanager.IReleaseManagerTypesArtifact
+		for i, opSetData := range opSetDataArray {
+			logger.Info("    Artifact %d:", i+1)
+			logger.Info("    Digest: %s", opSetData.Digest)
+			logger.Info("    Registry URL: %s", opSetData.RegistryUrl)
+
+			// this means this is performer
+			if opSetData.RegistryUrl == registryUrl {
+				err := updateContextWithDigest(opSetData.Digest)
+				if err != nil {
+					logger.Warn("Failed to update context with digest for operator set %s artifact %d: %v", opSetId, i+1, err)
+					continue
+				}
+				logger.Info("Successfully updated context with digest for operator set %s artifact %d", opSetId, i+1)
+			}
+
+			// Convert digest to bytes32
+			digestBytes, err := hexStringToBytes32(opSetData.Digest)
+			if err != nil {
+				logger.Warn("Failed to convert digest to bytes32 for operator set %s artifact %d: %v", opSetId, i+1, err)
+				continue
 			}
-			if cCtx.Bool("sign") {
-				log.Printf("Signing release artifacts...")
+
+			artifact := releasemanager.IReleaseManagerTypesArtifact{
+				Digest:      digestBytes,
+				RegistryUrl: opSetData.RegistryUrl,
 			}
+			artifacts = append(artifacts, artifact)
+		}
+
+		if len(artifacts) == 0 {
+			logger.Warn("No valid artifacts for operator set %s, skipping", opSetId)
+			continue
 		}
 
-		// Placeholder for future implementation
-		log.Printf("Release completed successfully")
+		logger.Info("Publishing release for operator set %s with %d artifacts...", opSetId, len(artifacts))
+		if err := publishReleaseToReleaseManagerAction(cCtx.Context, logger, avs, uint32(opSetIdInt), upgradeByTime, artifacts); err != nil {
+			if strings.Contains(err.Error(), "connection refused") {
+				logger.Warn("Failed to publish release for operator set %s: %v", opSetId, err)
+				logger.Info("Check if devnet is running and try again")
+				continue
+			}
```

Let's return an error if this happens. We don't want to continue.

---

https://github.com/Layr-Labs/devkit-cli/pull/199/commits/4a5950e3c9db1402208e6a0c253a3188383db1aa